### ISSUE TITLE
Fix request: False-positive AvoidRecompilingPatterns

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#89]
Fix false positive:
```
public class GoodComp {
    void otherCompile() {
        MyCompiler.compile("stuff");
    }
}

class MyCompiler {
    static void compile(String s) {
    }
}
```
use typeIs() to match fully qualified Pattern class.

**Comment by jborgers** [comment]
Fixed
